src/components/ScrollWithHeader.js

```diff
@@ -0,0 +1,106 @@
+import _ from 'lodash/fp'
+import { Fragment } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import * as Style from 'src/libs/style'
+import { Component } from 'src/libs/wrapped-components'
+import Interactive from 'react-interactive'
+
+
+export const emptyHeader = style => div({
+  style: _.merge({ color: 'transparent', userSelect: 'none' }, style)
+}, ['.'])
+
+
+export class ScrollWithHeader extends Component {
+  render() {
+    const { selectedTabIndex, header, children, negativeMargin, contentBackground } = this.props
+
+    return h(Fragment, [
+      div({
+        style: {
+          display: 'flex',
+          border: `1px solid ${Style.colors.sectionBorder}`,
+          borderBottom: 'unset',
+          backgroundColor: Style.colors.sectionHighlight,
+          borderTopRightRadius: 5,
+          borderTopLeftRadius: selectedTabIndex !== 0 && 5
+        }
+      }, [
+        div({ style: { padding: '0.5rem' } }, [
+          header
+        ])
+      ]),
+      div({
+        style: {
+          margin: negativeMargin && `0 -${negativeMargin}`,
+          borderTop: `2px solid ${Style.colors.secondary}`,
+          padding: negativeMargin && `0 ${negativeMargin}`,
+          backgroundColor: contentBackground
+        }
+      }, children)
+    ])
+  }
+}
+
+
+export class TabbedScrollWithHeader extends Component {
+  constructor(props) {
+    super(props)
+
+    this.controlled = props.selectedTabIndex !== undefined
```

Comment:
TOL: Be mindful that you're caching a prop here. In this case it's probably harmless, but something like this could just as well be computed on on the fly by a helper method